ladybug/datacollection.py

```diff
@@ -1,447 +1,438 @@
         """
-        data_points = self._data
-
-        values = [obj._value for obj in data_points]
-
-        lenght_values = len(values)
-
-        count = int(count)
-
-        assert count <= lenght_values, \
-            'Count must be equal to or smaller than list of data lenght'
-
-        assert count > 0, \
-            'Count must be higher than zero'
-
-        highest_values = sorted(values, reverse=True)[0:count]
-
-        highest_values_index = sorted(range(lenght_values),
-                                      key=lambda k: values[k],
-                                      reverse=True)[0:count]
-
-        return highest_values, highest_values_index
-
-    @staticmethod
-    def group_data_by_month(data, month_range=xrange(1, 13)):
-        """Return a dictionary of data points that are grouped for each month.
+        self._check_analysis_period(analysis_period)
+        _filtered_data = self.filter_by_moys(analysis_period.moys)
+        _filtered_data.header._analysis_period = analysis_period
+        return _filtered_data
 
-        Key values are between 1-12
+    def filter_by_hoys(self, hoys):
+        """Filter the Data Collection based on an analysis period.
 
         Args:
-            data: A list of DataPoint to be processed
-            month_range: A list of numbers for months. Default is 1-12
-        """
-        hourly_data_by_month = OrderedDict()
-        for m in month_range:
-            hourly_data_by_month[m] = []
-
-        for d in data:
-            try:
-                hourly_data_by_month[d.datetime.month].append(d)
-            except KeyError:
-                # month is not there
-                pass
-
-        return hourly_data_by_month
+           hoys: A List of hours of the year 0..8759
 
-    def group_by_month(self, month_range=xrange(1, 13)):
+        Return:
+            A new Data Collection with filtered data
         """
-        Return a dictionary of data points that are grouped for each month.
+        _moys = tuple(int(hour * 60) for hour in hoys)
+        return self.filter_by_moys(_moys)
 
-        Key values are between 1-12
+    def filter_by_moys(self, moys):
+        """Filter the Data Collection based on a list of minutes of the year.
 
         Args:
-           month_range: A list of numbers for months. Default is 1-12
-
-        Usage:
+           moys: A List of minutes of the year [0..8759 * 60]
 
-           epwfile = EPW("epw file address")
-           monthly_values = epwfile.dry_bulb_temperature.group_by_month()
-           print(monthly_values[2]) # returns values for the month of March
+        Return:
+            A new Data Collection with filtered data
         """
-        return self.group_data_by_month(self.data, month_range)
+        _filt_values, _filt_datetimes = self._filter_by_moys_slow(moys)
+        return HourlyDiscontinuousCollection(
+            self.header.duplicate(), _filt_values, _filt_datetimes)
```

Comment:
what happens to header in such cases?

Reply:
As I explained earlier.  Filtering by a patter, moys, or conditional statement does not edit the AnalysisPeriod in the header of the DataCollection.  This analysis period only changes when one filters by AnalysisPeriod and I think that this is the correct method.